lib/connection.h

```diff
@@ -753,5 +737,11 @@ namespace QMatrixClient
             static room_factory_t _roomFactory;
             static user_factory_t _userFactory;
     };
+
+    inline QDebug operator<<(QDebug dbg, const Connection::SupportedRoomVersion& v)
+    {
+        return dbg.nospace() << v.id << '/' << v.status;
```

Comment:
Did you actually try to run it without `QDebugStateSaver`? As far as I know `QDebug` is implicitly shared, so [copying it won't copy the underlying text stream](https://code.woboq.org/qt5/include/qt/QtCore/qdebug.h.html#_ZN6QDebugC1ERKS_), and [`nospace()` doesn't detach it either](https://code.woboq.org/qt5/include/qt/QtCore/qdebug.h.html#_ZN6QDebug7nospaceEv). To complicate matters, `printSequientialContainer()` does set `nospace()` already, but if you debug-print a single `SupportedRoomVersion`, you should get the difference.

Reply:
Yes, I actually tried.

The operator code:
```cpp
friend QDebug operator<<(QDebug dbg,
                         const SupportedRoomVersion& v)
{
    return dbg.nospace() << v.id << '/' << v.status;
}
```
I use the follow code to check:

```cpp
    QVector<Connection::SupportedRoomVersion> vec;
    vec.resize(2);
    vec[0].id = "my_id1";
    vec[0].status = "my_status1";
    vec[1].id = "my_id2";
    vec[1].status = "my_status2";
```
`qDebug().nospace() << vec << 123;` prints
```
QVector("my_id1"/"my_status1", "my_id2"/"my_status2")123
```

`qDebug() << vec << 123;` prints
```
QVector("my_id1"/"my_status1", "my_id2"/"my_status2") 123
```

Reply:
The `QDebugStateSaver` is indeed needed for single SupportedRoomVersion debug operator.

Reply:
Did you try `qDebug() << vec[0] << 123;`?

Reply:
Yes, this is how I figured out that the Saver is needed.